### PR TITLE
Collapse DB migrations that have already been released (pre-ATCv6)

### LINKED ISSUE #6027: Collapse database migrations for ATC 6
## I'm submitting a ...

-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...

-  Traffic Ops (database)

## Current behavior:

Our database migrations have not been collapsed since #3524 (first included in 4.0.0).

## New behavior:

We should squash database migrations in each major release, including the upcoming 6.0.0 release.

All migrations included in the 5.1.x branch should be collapsed, but migrations that will be new to the 6.0.x branch should remain uncollapsed.

### PR DESCRIPTION
**Pull request by shamrickus**
## What does this PR (Pull Request) do?

- [x] This PR fixes #6027 

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?
Run the `db/admin` tool on master and on this fork and compare the DB schema and data to ensure they are the same.
I used `pg_dump` to fetch the data from the DBs and `diff` to compare the output.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information
My diff:
```
> diff tops tops2
2685,2726c2685,2698
< 1	0	t	2021-07-28 08:22:43.182007
< 2	20181206000000	t	2021-07-28 08:22:43.185251
< 3	20190219000000	t	2021-07-28 08:22:43.186341
< 4	20190319000000	t	2021-07-28 08:22:43.187132
< 5	20190513000000	t	2021-07-28 08:22:43.188108
< 6	20191004000000	t	2021-07-28 08:22:43.191913
< 7	20191005000000	t	2021-07-28 08:22:43.195014
< 8	20191024000000	t	2021-07-28 08:22:43.202873
< 9	20191215000000	t	2021-07-28 08:22:43.206729
< 10	20200218000000	t	2021-07-28 08:22:43.207674
< 11	20200227000000	t	2021-07-28 08:22:43.209716
< 12	20200313000000	t	2021-07-28 08:22:43.210601
< 13	20200408000000	t	2021-07-28 08:22:43.212077
< 14	20200422101648	t	2021-07-28 08:22:43.215578
< 15	20200507000000	t	2021-07-28 08:22:43.231749
< 16	2020061622101648	t	2021-07-28 08:22:43.241121
< 17	2020062923101648	t	2021-07-28 08:22:43.24213
< 18	2020072700000000	t	2021-07-28 08:22:43.308803
< 19	2020081000000000	t	2021-07-28 08:22:43.311069
< 20	2020081108261100	t	2021-07-28 08:22:43.3147
< 21	2020082000000000	t	2021-07-28 08:22:43.315934
< 22	2020082700000000	t	2021-07-28 08:22:43.316662
< 23	2020103000000000	t	2021-07-28 08:22:43.319846
< 24	2020110200000000	t	2021-07-28 08:22:43.320716
< 25	2021010900000000	t	2021-07-28 08:22:43.322472
< 26	2021010900000001	t	2021-07-28 08:22:43.323647
< 27	2021010900000002	t	2021-07-28 08:22:43.327534
< 28	2021010900000003	t	2021-07-28 08:22:43.328402
< 29	2021012200000000	t	2021-07-28 08:22:43.331042
< 30	2021012700000000	t	2021-07-28 08:22:43.332001
< 31	2021012800000000	t	2021-07-28 08:22:43.333272
< 32	2021020500000000	t	2021-07-28 08:22:43.33405
< 33	2021021900000000	t	2021-07-28 08:22:43.334882
< 34	2021022300000000	t	2021-07-28 08:22:43.338818
< 35	2021030300000000	t	2021-07-28 08:22:43.342562
< 36	2021031400000000	t	2021-07-28 08:22:43.344308
< 37	2021040819481469	t	2021-07-28 08:22:43.349096
< 38	2021041200000000	t	2021-07-28 08:22:43.349876
< 39	2021052400000000	t	2021-07-28 08:22:43.351517
< 40	2021061100000000	t	2021-07-28 08:22:43.355434
< 41	2021070600000000	t	2021-07-28 08:22:43.35952
< 42	2021070800000000	t	2021-07-28 08:22:43.360362
---
> 1	0	t	2021-07-28 11:46:25.799758
> 2	2021012700000000	t	2021-07-28 11:46:25.802868
> 3	2021012800000000	t	2021-07-28 11:46:25.804771
> 4	2021020500000000	t	2021-07-28 11:46:25.805634
> 5	2021021900000000	t	2021-07-28 11:46:25.806486
> 6	2021022300000000	t	2021-07-28 11:46:25.810273
> 7	2021030300000000	t	2021-07-28 11:46:25.813885
> 8	2021031400000000	t	2021-07-28 11:46:25.815709
> 9	2021040819481469	t	2021-07-28 11:46:25.820572
> 10	2021041200000000	t	2021-07-28 11:46:25.821395
> 11	2021052400000000	t	2021-07-28 11:46:25.823174
> 12	2021061100000000	t	2021-07-28 11:46:25.826771
> 13	2021070600000000	t	2021-07-28 11:46:25.831382
> 14	2021070800000000	t	2021-07-28 11:46:25.832155
3105c3077
< SELECT pg_catalog.setval('public.goose_db_version_id_seq', 42, true);
---
> SELECT pg_catalog.setval('public.goose_db_version_id_seq', 14, true);
```

### CODE REVIEW

File: traffic_ops/app/db/create_tables.sql [hunk 1]
```diff
@@ -2424,6 +2738,121 @@ END IF;
...
+                    FOR EACH ROW
+                        EXECUTE PROCEDURE %s_last_updated(''%s'');
+                    ',
+                    QUOTE_IDENT(trigger_name),
+                    QUOTE_IDENT(table_name),
+                    QUOTE_IDENT(trigger_name),
+                    QUOTE_IDENT(table_name)
+                    );
+            END IF;
+            IF table_name = 'topology' THEN
```
Comment (shamrickus) [inline traffic_ops/app/db/create_tables.sql:2827]: Topology index is incorrect, see #6066

File: traffic_ops/app/db/create_tables.sql [hunk 2]
```diff
@@ -2424,6 +2738,121 @@ END IF;
...
+                EXECUTE FORMAT('
+                        CREATE INDEX IF NOT EXISTS %s_last_updated_idx
+                               ON %s_cachegroup (last_updated DESC NULLS LAST);
+                        ',
+                        QUOTE_IDENT(table_name),
+                        QUOTE_IDENT(table_name)
+                    );
+            ELSIF table_name = 'phys_location' THEN
```
Comment (shamrickus) [inline traffic_ops/app/db/create_tables.sql:2835]: phys_location index has a typo, see #6066

File: traffic_ops/app/db/create_tables.sql [hunk 3]
```diff
@@ -50,9 +50,99 @@ BEGIN
...
+--
+-- Name: on_delete_current_timestamp_last_updated(); Type: FUNCTION; Schema: public; Owner: traffic_ops
+--
+
+CREATE OR REPLACE FUNCTION public.on_delete_current_timestamp_last_updated()
+    RETURNS trigger
+AS $$
+BEGIN
+    update last_deleted set last_updated = now() where table_name = TG_ARGV[0];
+    RETURN NEW;
```
Comment (zrhoffman) [inline traffic_ops/app/db/create_tables.sql:139]: If these lines are unindented by 2 spaces, the diff

```diff
--- master.sql	2021-07-28 16:26:17.715951603 -0600
+++ shamrickus.sql	2021-07-28 16:25:47.495390677 -0600
@@ -243,8 +243,8 @@
     LANGUAGE plpgsql
     AS $$
 BEGIN
-  update last_deleted set last_updated = now() where table_name = TG_ARGV[0];
-  RETURN NEW;
+    update last_deleted set last_updated = now() where table_name = TG_ARGV[0];
+    RETURN NEW;
 END;
 $$;
 
```

disappears.

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Review by zrhoffman — APPROVED** [review]
Verified that
* `create_tables.sql` can still be run twice, and
* A DB dump with collapsed migrations has the same sha256 sum as a DB dump from `apache/master` without collapsed migrations:
  ```shell
  (set -o errexit -o nounset;
  create_migrate_dump() {(
      create_migrate_dump() {
        trap 'echo "Error on line ${LINENO} of ${0}" >/dev/stderr; exit 1' ERR;
        set -o errexit -o nounset -o pipefail;
        {
          docker-entrypoint.sh postgres >/dev/stderr &
          sleep 10;
          <create_tables.sql psql;
          <create_tables.sql psql;
          for migration in migrations/*; do
            <"$migration" sed '/+goose Down/,$d' | psql;
          done;
        } >/dev/stderr;
        pg_dump | sed '/^-- Dumped/d';
      };
      docker run --rm -iw/db \
        -v "$(pwd)/traffic_ops/app/db:/db" \
        -e PGUSER=traffic_ops \
        -e PGPASSWORD=twelve \
        -e POSTGRES_USER=traffic_ops \
        -e POSTGRES_PASSWORD=twelve \
        postgres:13-alpine bash -c "$(type create_migrate_dump | tail -n+2); create_migrate_dump";
  )};
  for remote_branch in apache/master shamrickus/to/6.0-migration; do
    git fetch "${remote_branch/\/*}";
    git checkout "${remote_branch}";
    create_migrate_dump >"${remote_branch/\/*}.sql";
  done;
  diff -u master.sql shamrickus.sql;
  openssl sha256 master.sql shamrickus.sql)
  ```

**Pull request merged**